source/common/upstream/outlier_detection_impl.h

```diff
@@ -99,50 +101,107 @@ class SuccessRateAccumulator {
   std::unique_ptr<SuccessRateAccumulatorBucket> backup_success_rate_bucket_;
 };
 
+class SuccessRateMonitor {
+public:
+  SuccessRateMonitor() = delete;
```

Comment:
nit: I don't think this is needed. It shouldn't compile if there is another constructor.

Reply:
Yes. Deleted that line.